python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,345 @@
+    params = {}
+    for key, value in properties.items():
+        if key == "comment":
+            database.description = value
+        elif key == "comment":
+            database.description = value
+        else:
+            params[key] = value
+    database.parameters = params
+    return database
+
+
+class HiveCatalog(Catalog):
+    _client: _HiveClient
+
+    @staticmethod
+    def identifier_to_database(identifier: Union[str, Identifier]) -> str:
+        tuple_identifier = Catalog.identifier_to_tuple(identifier)
+        if len(tuple_identifier) != 1:
+            raise ValueError("We expect a database name, Hive does not support hierarchical namespaces")
+
+        return tuple_identifier[0]
+
+    @staticmethod
+    def identifier_to_database_table(identifier: Union[str, Identifier]) -> Tuple[str, str]:
+        tuple_identifier = Catalog.identifier_to_tuple(identifier)
+        if len(tuple_identifier) != 2:
+            raise ValueError("We expect a database.table, Hive does not support hierarchical namespaces")
+
+        return tuple_identifier[0], tuple_identifier[1]
+
+    def __init__(self, name: str, properties: Properties, url: str):
+        super().__init__(name, properties)
+        self._client = _HiveClient(url)
+
+    def _convert_hive_into_iceberg(self, table: HiveTable) -> Table:
+        # Requires reading the manifest, will implement this in another PR
+        # Check the table type
+        return Table(identifier=(table.dbName, table.tableName))
+
+    def create_table(
+        self,
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: Optional[PartitionSpec] = None,
+        sort_order: SortOrder = UNSORTED_SORT_ORDER,
+        properties: Optional[Properties] = None,
+    ) -> Table:
+        database_name, table_name = self.identifier_to_database_table(identifier)
+        current_time_millis = int(time())
+        tbl = HiveTable(
+            dbName=database_name,
+            tableName=table_name,
+            owner=getuser(),
+            createTime=current_time_millis // 1000,
+            lastAccessTime=current_time_millis // 1000,
+            sd=_construct_hive_storage_descriptor(schema, location),
+            tableType="EXTERNAL_TABLE",
+            parameters=_construct_parameters("s3://"),
+        )
+        try:
+            with self._client as open_client:
+                open_client.create_table(tbl)
+                hive_table = open_client.get_table(dbname=database_name, tbl_name=table_name)
+        except AlreadyExistsException as e:
+            raise AlreadyExistsError(f"Table {database_name}.{table_name} already exists") from e
+        return self._convert_hive_into_iceberg(hive_table)
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        database_name, table_name = self.identifier_to_database_table(identifier)
```

Comment:
`identifer_to_database_table` will raise `ValueError`, but the right error for this context is `NoSuchTableError`. `ValueError` is okay, but callers of this API are likely to be able to handle `NoSuchTableError` rather than `ValueError`.

We will probably want an argument to control what is thrown. There are times when `ValueError` is correct, like when creating a table or for the `to` identifier in a rename. But places where the identifier is supposed to identify an existing table, I think we want to raise `NoSuchTableError.

Reply:
I made it configurable, and also added (and updated) the docstring. For me, it feels arbitrary and when you have faulty input, I would expect a `ValueError` even before it reaches out to Hive. The `NoSuchTableError` feels to me like a response from Hive. Anyway, with the docstrings, I hope it is clear to the end-user.

Reply:
This behavior is more in line with other implementations, which will throw `NoSuchTableException` if the namespace doesn't exist. The difference is that here, the namespace can't exist because it is invalid (so we know it doesn't exist). Does that make more sense?

Reply:
Yes it does 👍🏻